test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/condition/TreeHasItem.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2007-2019 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.condition;
+
+import org.eclipse.reddeer.common.condition.AbstractWaitCondition;
+import org.eclipse.reddeer.common.exception.RedDeerException;
+import org.eclipse.reddeer.swt.impl.tree.DefaultTreeItem;
+
+/**
+ * 
+ * @author jkopriva@redhat.com
+ * 
+ */
+public class TreeHasItem extends AbstractWaitCondition {
+	
+	String label;
+
+	public TreeHasItem(String label) {
+		this.label = label;
+	}
+
+	@Override
+	public boolean test() {
+		try {
```

Comment:
Not that nice solution to check existence of tree item in a tree. But I do not think of any other that would not throw an exception anyway. We might think it more in RD. otherwise it is fine.